pycode/test/test_epidata_modifyDataframeSeries.py

```diff
@@ -0,0 +1,90 @@
+#############################################################################
+# Copyright (C) 2020-2021 German Aerospace Center (DLR-SC)
+#
+# Authors:
+#
+# Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#############################################################################
+
+from datetime import datetime
+import unittest
+import numpy as np
+import pandas as pd
+import pandas.testing as pd_testing
+from unittest.mock import patch
+from pyfakefs import fake_filesystem_unittest
+from epidemiology.epidata import modifyDataframeSeries as mDfS
+
+
+class Test_modifyDataframeSeries(fake_filesystem_unittest.TestCase):
+    filldates_df = pd.DataFrame(
+        {'Date': ['2021-01-05', '2021-01-06', '2021-01-08'],
+         'test_col1': [11, 3, 6],
+         'test_col2': ['a', 'x', 't']})
+    test_df1 = pd.DataFrame(
+        {'Date': 
+        [datetime(2021, 1, 5),
+         datetime(2021, 1, 6),
+         datetime(2021, 1, 7),
+         datetime(2021, 1, 8)],
+         'test_col1': [11.0, 3.0, 3.0, 6.0],
+         'test_col2': ['a', 'x', 'x', 't']})
+    test_df2 = pd.DataFrame({
+        'Date':
+        [datetime(2021, 1, 4),
+         datetime(2021, 1, 5),
+         datetime(2021, 1, 6),
+         datetime(2021, 1, 7),
+         datetime(2021, 1, 8)],
+        'test_col1': [11.0, 11.0, 3.0, 3.0, 6.0],
+        'test_col2': ['a', 'a', 'x', 'x', 't']})
+    test_df3 = pd.DataFrame({
+        'Date':
+        [datetime(2021, 1, 4),
+         datetime(2021, 1, 5),
+         datetime(2021, 1, 6),
+         datetime(2021, 1, 7),
+         datetime(2021, 1, 8)],
+        'test_col1': [11.0, 11.0, 3.0, 3.0, 6.0],
+        'test_col2': ['a', 'a', 'x', 'x', 't']})
```

Comment:
The modify* one of the most important functions of the tool section and should be tested more!

You should have one short and one longer sequence (more than 7 days) and test that the moving average for e.g. 3 and 7 days for both frames is computed correctly.

Please note that smoothing only happens if all the values necessary are in the frame, if not the unsmoothed value (e.g. at the end) is taken itself.